File: back-end/src/main/java/com/tppartdeux/edoctorat/dto/professeur/SujetResponse.java
package com.tppartdeux.edoctorat.dto.professeur;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Sujet response DTO matching Angular's Sujet interface
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SujetResponse {
    private Long id;
    private ProfesseurResponse professeur;
    private ProfesseurResponse coDirecteur;
    private String titre;
    private String description;
    private FormationDoctoraleResponse formationDoctorale;
    private Boolean publier;
}

File: front-end/src/models/SujetResponse.ts
import { Professeur } from './Professeur';
import { FormationDoctorale } from './FormationDoctorale';

export interface SujetResponse {
  id: number;
  titre: string;
  description: string;
  motsCles: string;
  dateDepot: string;
  publier: boolean;
  pathFile: string;
  professeur: Professeur;
  formationDoctorale: FormationDoctorale;
  coDirecteur?: Professeur | null;  // Optional coDirecteur field from backend
}